src/Query/Filter/Field.ts

```diff
@@ -23,6 +23,9 @@ export abstract class Field {
     // Filtering
     // -----------------------------------------------------------------------------------------------------------------
 
+    // TODO add previous and next
+    protected static readonly currentPeriodRegexp = /^current (week|month|year)$/;
```

Comment:
> It seems a bit odd that a date-specific field is in the general abstract base class `Field`, rather than in `DateField`...

Totally agree, I have first put it in the DateField class but I need the reference in HappensDateField class which extends Field and not DateField...I preferred this rather than duplicate.

I may duplicate if that is your decision. Up to you.

Reply:
> I may duplicate if that is your decision. Up to you.

No, don't duplicate it. It's a `protected static` - just make it `public static` instead for now, then you can use it elsewhere.